### PR TITLE
feat: update vitest version v0.34 to V2.11

### PR DESCRIPTION
**Pull request by sean-brydon**
Updates vitest version from V0.34 to V2.11

`.toContains` has breaking changes - we now have to `.toEqual(expect.objectContaining(...))`

`.toContains` now only check if a array includes a value

image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -85,8 +86,10 @@
     "@playwright/test": "^1.45.3",
     "@snaplet/copycat": "^4.1.0",
     "@testing-library/jest-dom": "^5.16.5",
+    "@testing-library/react": "^16.0.1",
     "@types/jsdom": "^21.1.3",
     "@types/jsonwebtoken": "^9.0.3",
+    "@vitest/ui": "^2.1.1",
```
Comment (sean-brydon) [inline package.json:92]: Dev dep to get a lovely UI for running tests https://vitest.dev/guide/ui

File: apps/web/test/lib/next-config.test.ts
Comment (sean-brydon) [inline apps/web/test/lib/next-config.test.ts:1]: These dont need object containing as theyre exact matches

File: apps/web/test/utils/bookingScenario/test.ts [hunk 2]
```diff
@@ -16,14 +16,13 @@ const _testWithAndWithoutOrg = (
   const t = mode === "only" ? test.only : mode === "skip" ? test.skip : test;
   t(
     `${description} - With org`,
-    async ({ emails, sms, meta, task, onTestFailed, expect, skip }) => {
```
Comment (sean-brydon) [inline apps/web/test/utils/bookingScenario/test.ts:19]: Meta is no longer coming from the base vitest test -

File: tests/libs/__mocks__/prismaMock.ts [hunk 3]
```diff
@@ -14,5 +14,5 @@ beforeEach(() => {
   mockReset(prisma);
 });
 
-const prisma = mockDeep<PrismaClient>();
```
Comment (sean-brydon) [inline tests/libs/__mocks__/prismaMock.ts:17]: Weirdly - TS infering this was a massive performance hit. Casting it to the same type seemed to fix some of my issues

File: apps/web/test/lib/next-config.test.ts [hunk 4]
```diff
@@ -80,7 +80,8 @@ describe("next.config.js - Org Rewrite", () => {
 
   describe("Rewrite", () => {
     it("booking pages", () => {
-      expect(orgUserTypeRouteMatch("/user/type")?.params).toContain({
+      console.log(orgUserTypeRouteMatch("/user/type").params);
```
Comment (keithwillcode) [inline apps/web/test/lib/next-config.test.ts:83]: Still need this?

Comment (sean-brydon) [reply apps/web/test/lib/next-config.test.ts:83]: Oh no good spot!

Comment (sean-brydon) [reply apps/web/test/lib/next-config.test.ts]: Fixed https://github.com/calcom/cal.com/pull/16676/commits/adb09fbb633be1797c907a36933d288b8bc64e58

**Review by zomars — APPROVED** [review]

**Pull request merged**